Service/Menu/Cloner.php

```diff
@@ -79,30 +91,40 @@ public function clone(MenuInterface $menu): MenuInterface
             $this->menuIdentifierProcessor->getNewIdentifier($menu->getIdentifier())
         );
 
-        $this->menuRepository->save($menuClone);
-        $menuClone->saveStores($menu->getStores());
+        $connection = $this->resource->getConnection();
+        $connection->beginTransaction();
 
-        $menuCloneId = $menuClone->getId();
-        $nodeIdMap = [];
+        try {
+            $this->menuRepository->save($menuClone);
+            $menuClone->saveStores($menu->getStores());
 
-        foreach ($this->menuNodes->getList($menu) as $node) {
-            $nodeClone = $this->nodeFactory->create();
+            $menuCloneId = $menuClone->getId();
+            $nodeIdMap = [];
 
-            $nodeClone->setData($node->getData());
-            $nodeClone->setId(null);
-            $nodeClone->setMenuId($menuCloneId);
+            foreach ($this->menuNodes->getList($menu) as $node) {
+                $nodeClone = $this->nodeFactory->create();
 
-            if (isset($nodeIdMap[$node->getParentId()])) {
-                $nodeClone->setParentId($nodeIdMap[$node->getParentId()]);
-            }
+                $nodeClone->setData($node->getData());
+                $nodeClone->setId(null);
+                $nodeClone->setMenuId($menuCloneId);
 
-            if ($node->getImage()) {
-                $nodeClone->setImage($this->nodeImage->clone($node->getImage()));
-            }
+                if (isset($nodeIdMap[$node->getParentId()])) {
+                    $nodeClone->setParentId($nodeIdMap[$node->getParentId()]);
+                }
+
+                if ($node->getImage()) {
```

Comment:
suggestion for further development: based on node type call `cloneNode` in node type handler class so every node type can have specific cloning instructions (like here possibly handle duplication of uploaded image for specific node)